packages/popover2/src/popover2.md

```diff
@@ -0,0 +1,448 @@
+
+`Popover2` is built on top of the [**Popper.js**](https://popper.js.org) library.
+Popper.js is a small library that offers a powerful, customizable
+positioning engine and operates at blazing speed (`~60fps`).
+
+@reactExample Popover2Example
+
+@## Props
+
+`Popover` supports controlled and uncontrolled usage through `isOpen` and
+`defaultIsOpen`, respectively. Use `onInteraction` in controlled mode to respond
+to changes in the `isOpen` state.
+
+Supported user interactions are dictated by the `interactionKind` prop.
+
+This component is quite powerful and has a wide range of features. Explore the
+[**Concepts**](#core/components/popover.concepts) section below for more advanced
+guides.
+
+@interface IPopover2Props
+
+@## Concepts
+
+@### Structure
+
+When creating a popover, you must specify both its **content** (via the `content` prop) and
+its **target** (via the `renderTarget` prop).
+
+The **target** is rendered at the location of the Popover2 component in the React component tree. It acts
+as the trigger for the popover; user interaction will show the popover based on the `interactionKind` prop.
+In Popper.js terms, this is the popper "reference". In order to add its interaction logic to the target,
+Popover2 supplies an object of props to the `renderTarget` function. These props should be
+[spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals)
+out to the `JSX.Element` returned from `renderTarget`.
+
+The **content** will be shown inside the popover itself. When opened, the popover will always be
+positioned on the page next to the target; the `position` prop determines its relative position (on
+which side of the target).
+
+<div class="@ns-callout @ns-intent-warning @ns-icon-warning-sign">
+    <h4 class="@ns-heading">Button targets</h4>
+
+Buttons make great popover targets, but the `disabled` attribute on a `<button>` blocks all
+events, which interferes with the popover functioning. If you need to disable a button which
+triggers a popover, you should use [`AnchorButton`](#core/components/button.anchor-button) instead.
+See the [callout here](#core/components/button.props) for more details.
+
+</div>
+
+```tsx
+import { Button } from "@blueprintjs/core";
+import { Popover2 } from "@blueprintjs/popover2";
+
+export class PopoverExample extends React.PureComponent {
+    public render() {
+        // popover content gets no padding by default; add the "@ns-popover-content-sizing"
+        // class to the popover to set nice padding between its border and content.
+        return (
+            <Popover2
+                interactionKind="click"
+                popoverClassName="@ns-popover-content-sizing"
+                placement="bottom"
+                content={
+                    <div>
+                        <h5>Popover title</h5>
+                        <p>...</p>
+                        <Button className="@ns-popover-dismiss">Dismiss</Button>
```

Comment:
nit: use `Classes.POPOVER2_DISMISS` here, and also the `text={}` prop for Button